components/hab/src/cli.rs

```diff
@@ -1326,6 +1331,14 @@ fn file_exists_or_stdin(val: String) -> result::Result<(), String> {
     }
 }
 
+#[allow(clippy::needless_pass_by_value)] // Signature required by CLAP
+fn valid_ipv4_address(val: String) -> result::Result<(), String> {
+    match Ipv4Addr::from_str(&val) {
+        Ok(_) => Ok(()),
+        Err(_) => Err(format!("'{}' is not a valid IPv4 address, eg: '0.0.0.0'", val)),
```

Comment:
I can't decide if this is a problem worth fixing or not, but my concern with this particular error message is it might lead people who receive it to pass `--sys-ip-address 0.0.0.0` to the supervisor, which, while a valid IP address, is not a good choice for `sys.ip`.  I might consider changing it to something else, perhaps a random address from one of the reserved IP blocks: `10.0.0.0/8`, `192.168.0.0/16`, or `172.16.0.0/12`.

Reply:
Great point! Changed in [188452](https://github.com/habitat-sh/habitat/pull/6960/commits/1884529be226c3eb326c13adb2c70a1e49accd87).